### PR TITLE
Review c10

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Reckon this is good to merge. Please review and merge @csgillespie

File: 10-learning.Rmd [hunk 1]
```diff
@@ -19,27 +19,26 @@ help, reading R source code, and use of online material.
...
 
-The main disadvantage of R's internal help is that it is notoriously terse and in some cases sparse. So do not expect to *always* be able to find the answer in R: while we recommend R's internal help is the first place to look, we do not recommend *relying* on it. From a learning perspective becoming aquainted with R's documentation is often better than finding out the solution from a different source: by reading the documentation, you will see what the function was designed to do and perhaps gain further insight about its history, e.g. through references to academic papers.
+The main disadvantage of R's internal help is that it is notoriously terse and in some cases sparse. So do not expect to *always* be able to find the answer in R: while we recommend R's internal help is the first place to look, we do not recommend *relying* on it. From a learning perspective becoming acquainted with R's documentation is often better than finding out the solution from a different source: by reading the documentation, you will see what the function was designed to do and perhaps gain further insight about its history, e.g. through references to academic papers.
```
Comment (csgillespie) [inline 10-learning.Rmd]: Base R docs is aimed at developers not users. Also, you wouldn't expect the mean function to explain what the mean is.

File: 10-learning.Rmd [hunk 2]
```diff
@@ -68,12 +67,13 @@ help.search(pattern = "optimisation|optimization", fields = c("title", "concept"
...
 
-```{r, eval=FALSE}
+```{r}
 apropos("optim")
-# Showing the first six results
-apropos("lm")[1:6]
+apropos("lm")[1:6] # show only first six results
 ```
 
+To search *all R packages*, including those you have not installed locally, for a specific topic there are a number of options. For obvious reasons, this depends on having internet access. The most rudimentary way to see what packages are available from CRAN, if you are using RStudio, is to use its autocompletion functionality for package names. To take and example, if you are looking for a package for geospatial data analysis, you could do worse than enter the text string "geo" (without quote marks) as an argument into `install.packages()` and hitting `Enter`. The results are are shown in the figure below.
```
Comment (csgillespie) [inline 10-learning.Rmd]: Do you mean  install.packages(geo)?

File: 10-learning.Rmd [hunk 3]
```diff
@@ -102,34 +102,63 @@ edit(v)
...
 
 ```{r eval=FALSE}
-# help("plot") # or, more concisely:
-?plot
+# help("optim") # or, more concisely:
+?optim
 ```
 
-In general, help pages describe the public API of the package. This is one of the reasons
+In general, help pages describe *what* functions do, not *how* they work. This is one of the reasons
```
Comment (csgillespie) [inline 10-learning.Rmd]: Much better explanation.

File: 10-learning.Rmd [hunk 4]
```diff
@@ -203,14 +245,22 @@ many other special interest mailing [lists](https://www.r-project.org/mail.html)
...
 list, read the relevant mailing archive and check that your message is appropriate.
 
-### Asking a question
+## Asking a question
+
+A great way to get specific help on a difficult topic is to ask for help.
+However, asking a good question is not easy. Three common mistakes, and ways to avoid them, outlined below:
 
-Asking questions on stackoverflow and R-help is hard. Your question should contain
+1. Asking a question that has already been asked: ensure you've properly searched for the answer before posting.
```
Comment (csgillespie) [inline 10-learning.Rmd]: Excellent list!

File: 10-learning.Rmd [hunk 5]
```diff
@@ -239,13 +289,39 @@ dput(example_df)
...
+1. Free webinars and online courses provided by [RStudio](http://www.rstudio.com/resources/webinars/) and [DataCamp](https://www.datacamp.com/community/open-courses). Both organisations are well regarded and keep their content up-to-date, but there are likely other sources of other online courses. We recommend testing pushing your abilities, rather than going over the same material covered in this book. 
+1. *R for Data Science* [@grolemund_r_2016], a free book introducing many concepts and 'tidy' packages for working with data (a free online version is available from [r4ds.had.co.nz/](http://r4ds.had.co.nz/)).
+1. *R programming for Data Science* [@peng_r_2014], which provides in depth coverage of analysis and visualisation of datasets.  
+1. *Advanced R Programming* [@Wickham2014], an advanced book which looks at the internals of how R works (free from [adv-r.had.co.nz](http://adv-r.had.co.nz/)).
+
+## Spread the knowledge
+
+The final thing to say on the topic of efficient learning relates to the [old](https://en.wikipedia.org/wiki/Docendo_discimus) (~2000 years old!) saying *docendo discimus*:
+
+> **by teaching we learn**.
```
Comment (csgillespie) [inline 10-learning.Rmd]: Nice!

File: 10-learning.Rmd (hunk 1 above)
Comment (Robinlovelace) [inline 10-learning.Rmd]: Sorted in latest commit - now shorter and solves this issue I think.

File: 10-learning.Rmd (hunk 2 above)
Comment (Robinlovelace) [inline 10-learning.Rmd]: Yes - figure added

**Comment by Robinlovelace** [comment]
@csgillespie  :+1: great stuff, glad you like it, ready to merge I'd say.

**Pull request merged**